src/main/scala/org/broadinstitute/dsde/firecloud/webservice/AgoraEntityApiService.scala

```diff
@@ -0,0 +1,33 @@
+package org.broadinstitute.dsde.firecloud.webservice
+
+import org.broadinstitute.dsde.firecloud.model.MethodRepository.EditMethodRequest
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.impEditMethodRequest
+import org.broadinstitute.dsde.firecloud.service.{AgoraEntityService, FireCloudDirectives, FireCloudRequestBuilding}
+import org.broadinstitute.dsde.firecloud.utils.StandardUserInfoDirectives
+import org.broadinstitute.dsde.firecloud.model.UserInfo
+import spray.httpx.SprayJsonSupport._
+import spray.routing.{HttpService, Route}
+
+import scala.concurrent.ExecutionContext
+
+trait AgoraEntityApiService extends HttpService with FireCloudRequestBuilding with FireCloudDirectives
+  with StandardUserInfoDirectives{
+
+  private implicit val ec: ExecutionContext = actorRefFactory.dispatcher
+
+  val agoraEntityServiceConstructor: UserInfo => AgoraEntityService
+
+  val agoraEntityRoutes: Route = {
+    path("methods" / "edit") {
+      requireUserInfo() { userInfo =>
+        entity(as[EditMethodRequest]) { editMethodRequest =>
```

Comment:
This requires that the end user specify a fully-populated `EditMethodRequest`, which contains a synopsis, documentation, and payload. Since we are editing a pre-existing method, I think it would make more sense to make synopsis, documentation, and payload optional, and copy them from the pre-existing method if the user didn't specify them.

Similarly, let's make `redactOldSnapshot` optional and give it a default of false.